mijia-homie/src/main.rs

```diff
@@ -289,19 +295,20 @@ async fn bluetooth_connection_loop(
 
         // Check the state of each sensor and act on it if appropriate.
         {
-            let ids: Vec<DeviceId> = state.lock().await.sensors.keys().cloned().collect();
-            for id in ids {
+            let mac_addresses: Vec<MacAddress> =
+                state.lock().await.sensors.keys().cloned().collect();
```

Comment:
(maybe later) would be fun to use a MacAddress type that is Copy and have this be `state.lock().await.sensors.keys().copied().collect()`

Reply:
Having `MacAddress`, which is essentially a string, be `Copy` feels wrong to me.